src/xdsl/dialects/affine.py

```diff
@@ -1,8 +1,13 @@
+from __future__ import annotations
 from typing import Union
+from enum import Enum
+from pyrsistent import optional
```

Comment:
Is there a reason to use the `pyrsistent` `optional` rather than `Optional` from `typing`